File: app/src/main/java/nl/rsdt/japp/jotial/data/bodies/AutoPostBody.java
package nl.rsdt.japp.jotial.data.bodies;

import com.google.gson.annotations.SerializedName;

import nl.rsdt.japp.application.JappPreferences;

public class AutoPostBody {

    @SerializedName("SLEUTEL")
    private String SLEUTEL;

    @SerializedName("gebruikersID")
    private String id;

    @SerializedName("gebruikersNaam")
    private String gebruikersNaam;

    @SerializedName("autoEigenaar")
    private String autoEigenaar;

    @SerializedName("rol")
    private String rol;

    @SerializedName("taak")
    private String taak;

    public void setSleutel(String sleutel) {
        this.SLEUTEL = sleutel;
    }

    public void setId(int id) {
        this.id = String.valueOf(id);
    }

    public void setGebruikersNaam(String gebruikersNaam) {
        this.gebruikersNaam = gebruikersNaam;
    }

    public void setAutoEigenaar(String autoEigenaar) {
        this.autoEigenaar = autoEigenaar;
    }

    public void setRol(String rol) {
        this.rol = rol;
    }

    public void setTaak(String taak) {
        this.taak = taak;
    }

    public static AutoPostBody getDefault()
    {
        AutoPostBody builder = new AutoPostBody();
        builder.setGebruikersNaam(JappPreferences.getAccountUsername());
        builder.setId(JappPreferences.getAccountId());
        builder.setRol("rol");
        builder.setTaak("taak");
        builder.setAutoEigenaar(JappPreferences.getAccountUsername());
        builder.setSleutel(JappPreferences.getAccountKey());
        return builder;
    }
}

File: app/src/main/java/nl/rsdt/japp/jotial/data/bodies/AutoPostBody.kt
package nl.rsdt.japp.jotial.data.bodies

import com.google.gson.annotations.SerializedName

import nl.rsdt.japp.application.JappPreferences

class AutoPostBody {

    @SerializedName("SLEUTEL")
    private var SLEUTEL: String? = null

    @SerializedName("gebruikersID")
    private var id: String? = null

    @SerializedName("gebruikersNaam")
    private var gebruikersNaam: String? = null

    @SerializedName("autoEigenaar")
    private var autoEigenaar: String? = null

    @SerializedName("rol")
    private var rol: String? = null

    @SerializedName("taak")
    private var taak: String? = null

    fun setSleutel(sleutel: String?) {
        this.SLEUTEL = sleutel
    }

    fun setId(id: Int) {
        this.id = id.toString()
    }

    fun setGebruikersNaam(gebruikersNaam: String?) {
        this.gebruikersNaam = gebruikersNaam
    }

    fun setAutoEigenaar(autoEigenaar: String?) {
        this.autoEigenaar = autoEigenaar
    }

    fun setRol(rol: String) {
        this.rol = rol
    }

    fun setTaak(taak: String) {
        this.taak = taak
    }

    companion object {

        val default: AutoPostBody
            get() {
                val builder = AutoPostBody()
                builder.setGebruikersNaam(JappPreferences.accountUsername)
                builder.setId(JappPreferences.accountId)
                builder.setRol("rol")
                builder.setTaak("onbekend")
                builder.setAutoEigenaar(JappPreferences.accountUsername)
                builder.setSleutel(JappPreferences.accountKey)
                return builder
            }
    }
}
